src/components/Tasks/Tasks.tsx

```diff
@@ -1,39 +1,73 @@
-import React, { useContext } from 'react'
+import React, { useContext, useState } from 'react'
 import { TasksContext } from '../../contexts/TasksContext'
 import { useTasks } from '../../hooks/useTasks'
 import { Loader } from '../Loader'
 import { TasksHeader } from '../TasksHeader'
 import { TasksList } from '../TasksList'
 import { TasksPending } from '../TasksPending'
+import classNames from 'classnames'
 
-export const Tasks = ({ tasksHeader }: { tasksHeader?: string }) => {
+export const Tasks = ({
+  tasksHeader,
+  collapsable = false,
+  defaultCollapsed = false,
+}: {
+  tasksHeader?: string
+  collapsable?: boolean
+  defaultCollapsed?: boolean
+}) => {
   const contextData = useTasks()
 
   return (
     <TasksContext.Provider value={contextData}>
       <div className='Layer__tasks-component'>
-        <TasksComponent tasksHeader={tasksHeader} />
+        <TasksComponent
+          tasksHeader={tasksHeader}
+          collapsable={collapsable}
+          defaultCollapsed={defaultCollapsed}
+        />
       </div>
     </TasksContext.Provider>
   )
 }
 
-export const TasksComponent = ({ tasksHeader }: { tasksHeader?: string }) => {
+export const TasksComponent = ({
+  tasksHeader,
+  collapsable = false,
+  defaultCollapsed = false,
+}: {
+  tasksHeader?: string
+  collapsable?: boolean
+  defaultCollapsed?: boolean
+}) => {
   const { isLoading, data } = useContext(TasksContext)
+  const [open, setOpen] = useState(defaultCollapsed ? false : true)
```

Comment:
I'd love to have the logic be: 

- If all tasks are complete: then default collapsed
- If there are any incomplete tasks, then default open
- This behavior can be overridden from the parent component

I _think_ we could solve this with this
```suggestion
  const [open, setOpen] = useState(defaultCollapsed ? data.tasks.filter(t => isComplete(t)).length = data.tasks.length)
```

Reply:
I had to add a bit more code, because `data` is empty at first painting. Change commited, feel free to merge PR